datalake/src/main/java/com/sequenceiq/datalake/flow/upgrade/ccm/UpgradeCcmActions.java

```diff
@@ -0,0 +1,127 @@
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmFailedEvent;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmStackEvent;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmStackRequest;
+import com.sequenceiq.datalake.flow.upgrade.ccm.event.UpgradeCcmSuccessEvent;
+import com.sequenceiq.datalake.metric.MetricType;
+import com.sequenceiq.datalake.metric.SdxMetricService;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+import com.sequenceiq.flow.core.FlowEvent;
+import com.sequenceiq.flow.core.FlowParameters;
+import com.sequenceiq.flow.core.FlowState;
+
+@Configuration
+public class UpgradeCcmActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpgradeCcmActions.class);
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Inject
+    private SdxMetricService metricService;
+
+    @Bean(name = "UPGRADE_CCM_UPGRADE_STACK_STATE")
+    public Action<?, ?> upgradeCcmStack() {
+        return new AbstractSdxAction<>(UpgradeCcmStackEvent.class) {
+
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    UpgradeCcmStackEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, UpgradeCcmStackEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Execute CCM upgrade stack flow for SDX: {}", payload.getResourceId());
+                sendEvent(context);
+            }
+
+            @Override
+            protected Selectable createRequest(SdxContext context) {
+                return new UpgradeCcmStackRequest(context.getSdxId(), context.getUserId());
+            }
+
+            @Override
+            protected Object getFailurePayload(UpgradeCcmStackEvent payload, Optional<SdxContext> flowContext, Exception ex) {
+                return UpgradeCcmFailedEvent.from(payload, ex);
+            }
+        };
+    }
+
+    @Bean(name = "UPGRADE_CCM_FINISHED_STATE")
+    public Action<?, ?> finishedAction() {
+        return new AbstractSdxAction<>(UpgradeCcmSuccessEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    UpgradeCcmSuccessEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, UpgradeCcmSuccessEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("CCM upgrade finalized: {}", payload.getResourceId());
```

Comment:
Nit: Insert `for SDX` after `finalized`.

Reply:
ok